### PR TITLE
Added entity config types.

### PR DESCRIPTION
**Pull request by Alkesst**

### CODE REVIEW

**Comment by Alkesst** [comment]
Do not merge this PR until #1 is completed and merged to master.

**Comment by Alkesst** [comment]
Closing until Entity structure is properly defined.

**Comment by Alkesst** [comment]
Ready to be reviewed @samueljurado12

**Review by samueljurado12 — APPROVED** [review]

**Pull request merged**
